driver/vhci/vhci_read.c

```diff
@@ -672,6 +672,34 @@ store_urbr(PIRP irp, struct urb_req *urbr)
 	return status;
 }
 
+static VOID
+on_peding_irp_read_cancelled(PDEVICE_OBJECT devobj, PIRP irp_read)
+{
+	UNREFERENCED_PARAMETER(devobj);
+	DBGI(DBG_READ, "pending irp read cancelled");
+
+	KIRQL	oldirql;
+	PIO_STACK_LOCATION	irpstack;
+	pusbip_vpdo_dev_t	vpdo;
+
+	irpstack = IoGetCurrentIrpStackLocation(irp_read);
+	vpdo = irpstack->FileObject->FsContext;
```

Comment:
How about replacing with vpdo assignment in cancel_urbr()?
https://github.com/cezanne/usbip-win/blob/141e09a4447a86ef2a5a7f2f8fe8468104dfba55/driver/vhci/usbreq.c#L138

Reply:
good idea, done